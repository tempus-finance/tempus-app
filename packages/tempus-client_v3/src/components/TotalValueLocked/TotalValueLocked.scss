.tc__totalValueLocked {
 display: flex;
 flex-direction: column;
 align-items: flex-end;
 justify-content: flex-start;
}
