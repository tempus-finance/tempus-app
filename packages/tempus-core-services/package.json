{
  "name": "tempus-core-services",
  "version": "0.0.1",
  "private": true,
  "description": "Tempus core Services",
  "keywords": [
    "typescript",
    "boilerplate",
    "esbuild"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "lint": "eslint src/ --ext .js,.jsx,.ts,.tsx",
    "test": "jest",
    "clean": "rm -rf dist build package",
    "ts-node": "ts-node",
    "docs": "typedoc --entryPoints src/index.ts",
    "build": "npm run clean && tsc -p tsconfig.json",
    "build-all": "npm run clean && npm run build && npm run esbuild-node && npm run esbuild-node",
    "esbuild-node": "esbuild src/index.ts --bundle --platform=node --minify --sourcemap=external --outfile=dist/esbuild/index.js",
    "esbuild-node:dev": "esbuild src/index.ts --bundle --platform=node --sourcemap=external --outfile=dist/esbuild/index.js",
    "esbuild-node:watch": "esbuild src/index.ts --bundle --platform=node --watch --sourcemap=external --outfile=dist/esbuild/index.js"
  },
  "dependencies": {
    "@ethersproject/providers": "^5.5.2",
    "@types/jest": "^24.9.1",
    "@types/node": "^12.20.15",
    "axios": "^0.22.0",
    "date-fns": "^2.28.0",
    "ethers": "^5.4.6"
  },
  "devDependencies": {
    "esbuild": "^0.11.11",
    "eslint": "^7.22.0",
    "jest": "^26.6.3",
    "ts-jest": "^26.5.4",
    "ts-node": "^9.1.1",
    "typedoc": "^0.20.35"
  },
  "peerDependencies": {
    "@typescript-eslint/eslint-plugin": "^4.28.1",
    "@typescript-eslint/parser": "^4.28.1",
    "eslint-config-airbnb-typescript": "^12.3.1"
  }
}
