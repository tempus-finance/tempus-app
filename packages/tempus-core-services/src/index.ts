import ERC20ABI from './abi/ERC20.json';
import VaultABI from './abi/Vault.json';

export { ERC20ABI, VaultABI };

export type { ERC20 } from './abi';

export type {
  AvailableToDeposit,
  Chain,
  ChainConfig,
  Config,
  DynamicTempusPool,
  PoolShares,
  ProtocolDisplayName,
  ProtocolName,
  Ticker,
  NativeTokenTicker,
  TempusPool,
  TokenPrecision,
  TokenTypePrecision,
  YearnData,
  YearnDataApy,
  YearnDataToken,
} from './interfaces';

export type {
  DepositedEvent,
  ERC20TokenService,
  PoolBalanceChangedEvent,
  PoolBalanceChangedEventListener,
  RedeemedEvent,
  TempusAMMJoinKind,
  TempusAMMExitKind,
  TempusAMMService,
  TempusControllerService,
  TempusPoolService,
  TransferEventListener,
  VaultService,
  StatisticsService,
  StorageService,
  SwapEvent,
} from './services';

export type { Numberish } from './datastructures';

export { Decimal, ZERO, ONE, DEFAULT_DECIMAL_PRECISION } from './datastructures';

export { SwapKind } from './services';

export {
  getChainlinkFeed,
  getCoingeckoRate,
  getDefaultProvider,
  getERC20TokenService,
  initServices,
  getServices,
  getDefinedServices,
} from './services';

export {
  NumberUtils,
  DecimalUtils,
  capitalize,
  chainIdToChainName,
  chainIdHexToChainName,
  chainNameToHexChainId,
  chainToTicker,
  decreasePrecision,
  div18f,
  getCookie,
  getLibrary,
  getPastDaysNumber,
  getProvider,
  getProviderFromSignerOrProvider,
  getRangeFrom,
  getTokenPrecision,
  increasePrecision,
  isZeroString,
  mul18f,
  prettifyChainName,
  prettifyChainNameLong,
  rayToDai,
  reverseString,
  shortenAccount,
  wadToDai,
  wait,
  prettifyProtocolName,
} from './utils';

export * from './constants';
