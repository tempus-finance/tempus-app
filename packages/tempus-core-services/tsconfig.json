{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,

    "importHelpers": true,
    "experimentalDecorators": true,
    "sourceMap": true,
    "outDir": "./dist/",
    "types": ["node", "jest"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules"]
}
