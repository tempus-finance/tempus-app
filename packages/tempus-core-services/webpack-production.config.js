const path = require('path');
const webpack = require('webpack');

require('dotenv').config({ path: './.env.production' });

module.exports = {
  entry: './src/index.ts',
  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: [/node_modules/],
      },
    ],
  },
  resolve: {
    extensions: ['.ts', '.js'],
  },
  output: {
    path: path.resolve(__dirname, 'dist'),
    publicPath: '',
    filename: 'index.js',
    library: 'tempus-core-services',
    libraryTarget: 'umd',
  },
  mode: 'production',
  devtool: 'eval-source-map',
  externals: ['ethers', 'rxjs', 'date-fns', '@ethersproject/providers'],
  plugins: [
    new webpack.DefinePlugin({
      'process.env': JSON.stringify(process.env),
    }),
  ],
};
