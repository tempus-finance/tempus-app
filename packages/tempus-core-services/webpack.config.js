// Webpack uses this to work with directories
const path = require('path');
const webpack = require('webpack');

require('dotenv').config({ path: '../tempus-client_v3/.env.development' });

// This is the main configuration object.
// Here, you write different options and tell Webpack what to do
module.exports = {
  // Path to your entry point. From this file Webpack will begin its work
  entry: './src/index.ts',

  module: {
    rules: [
      {
        test: /\.tsx?$/,
        use: 'ts-loader',
        exclude: [/node_modules/],
        include: [path.resolve(__dirname, 'src')],
      },
    ],
  },
  resolve: {
    extensions: ['.ts', '.js'],
  },

  // Path and filename of your result bundle.
  // Webpack will bundle all JavaScript into this file
  output: {
    path: path.resolve(__dirname, 'dist'),
    publicPath: '',
    filename: 'index.js',
    library: 'tempus-core-services',
    libraryTarget: 'umd',
  },

  // Default mode for Webpack is production.
  // Depending on mode Webpack will apply different things
  // on the final bundle. For now, we don't need production's JavaScript
  // minifying and other things, so let's set mode to development
  mode: 'development',

  devtool: 'eval-source-map',

  watch: true,

  plugins: [
    new webpack.DefinePlugin({
      'process.env': JSON.stringify(process.env),
    }),
  ],
};
