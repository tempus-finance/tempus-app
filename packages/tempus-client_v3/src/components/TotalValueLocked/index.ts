import TotalValueLocked from './TotalValueLocked';

export default TotalValueLocked;
