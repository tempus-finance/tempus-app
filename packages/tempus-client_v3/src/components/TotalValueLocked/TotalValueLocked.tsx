import { FC, memo } from 'react';
import { useTranslation } from 'react-i18next';
import { DecimalUtils } from 'tempus-core-services';
import { useTotalTvl } from '../../hooks';
import { Typography } from '../shared';

import './TotalValueLocked.scss';

const TotalValueLocked: FC = () => {
  const { t } = useTranslation();
  const tvl = useTotalTvl();

  return (
    <div className="tc__totalValueLocked">
      <Typography variant="body-secondary" color="text-primary-inverted">
        {t('TotalValueLocked.title')}
      </Typography>
      <Typography variant="subtitle" color="text-primary-inverted" type="mono">
        {tvl && DecimalUtils.formatToCurrency(tvl, 2, '$')}
      </Typography>
    </div>
  );
};

export default memo(TotalValueLocked);
