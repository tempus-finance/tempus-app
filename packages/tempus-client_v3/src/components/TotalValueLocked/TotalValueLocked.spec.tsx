import { render } from '@testing-library/react';
import { Decimal } from 'tempus-core-services';
import TotalValueLocked from './TotalValueLocked';

const mockTvlValue = new Decimal('12345678');

jest.mock('../../hooks/useTvlData', () => ({
  useTvlData: () => [mockTvlValue],
}));

const subject = () => render(<TotalValueLocked />);

describe('TotalValueLockedfrom', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  it('renders a component that displays TVL value', () => {
    const { container, getByText } = subject();

    const component = container.querySelector('.tc__totalValueLocked');
    const tvl = getByText(/\$12,345,678.00/);

    expect(component).not.toBeNull();
    expect(tvl).not.toBeNull();
    expect(component).toMatchSnapshot();
  });
});
