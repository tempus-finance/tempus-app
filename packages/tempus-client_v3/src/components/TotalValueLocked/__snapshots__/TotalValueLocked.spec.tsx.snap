// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalValueLockedfrom renders a component that displays TVL value 1`] = `
<div
  class="tc__totalValueLocked"
>
  <div
    style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
  >
    Total Value Locked
  </div>
  <div
    style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(255, 255, 255); font-weight: 400; font-family: 'Azeret Mono', monospace; opacity: 1;"
  >
    $12,345,678.00
  </div>
</div>
`;
